src/client/sound/s_sample.c

```diff
@@ -34,41 +34,55 @@ static _Bool S_LoadSampleChunkFromPath(s_sample_t *sample, char *path, const siz
 
 	void *buf;
 	int32_t i;
-	SDL_RWops *rw;
 
 	buf = NULL;
-	rw = NULL;
 
 	i = 0;
 	while (SAMPLE_TYPES[i]) {
 
 		StripExtension(path, path);
-		g_strlcat(path, SAMPLE_TYPES[i++], pathlen);
+		g_strlcat(path, SAMPLE_TYPES[i], pathlen);
+
+		const char *ext = SAMPLE_TYPES[i] + 1;
+
+		i++;
 
 		int64_t len;
 		if ((len = Fs_Load(path, &buf)) == -1) {
 			continue;
 		}
 
-		if (!(rw = SDL_RWFromMem(buf, (int32_t) len))) {
-			Fs_Free(buf);
-			continue;
-		}
+		Sound_AudioInfo desired = {
+			.format = AUDIO_S16,
+			.channels = 1,
+			.rate = s_rate->integer
+		};
 
-		if (!(sample->chunk = Mix_LoadWAV_RW(rw, false))) {
-			Com_Warn("%s\n", Mix_GetError());
+		Sound_Sample *snd = Sound_NewSampleFromMem(buf, (Uint32) len, ext, &desired, 2048);
+		
+		if (!snd) {
+			Com_Warn("%s\n", Sound_GetError());
+		} else {
+			Uint32 snd_length = Sound_DecodeAll(snd);
```

Comment:
I usually try to use the same return value as the function I'm calling, not one that we use, just to ensure that we're always getting the right data back in case our own versions change for whatever reason. Personal preference I guess.